c34568783.lua

```diff
@@ -68,27 +69,31 @@ end
 function s.thtg(e,tp,eg,ep,ev,re,r,rp,chk,chkc)
 	local c=e:GetHandler()
 	if chkc then return chkc:IsLocation(LOCATION_GRAVE) and chkc:IsControler(tp) and s.tgfilter(chkc) and chkc~=c end
-	if chk==0 then return Duel.IsExistingTarget(s.tgfilter,tp,LOCATION_GRAVE,0,1,c) and c:IsAbleToHand() end
+	if chk==0 then return Duel.IsExistingTarget(s.tgfilter,tp,LOCATION_GRAVE,0,1,c)
+		and c:IsAbleToHand() and c:IsCanBeEffectTarget(e) end
```

Comment:
Is the usage of Duel.IsExistingTarget() and Card.IsCanBeEffectTarget() redundant?